### PR TITLE
[14.0] [MIG-IMP] delivery_ups_oca, migration and oauth

### PR DESCRIPTION
**Pull request by manuelregidor**
Working features:
- Token creation/renew
- Shipping rating
- New shipping
- Automatic label when a shipping is created
- Label recreation
- URL to UPS tracking website
- Shipping cancel
- Tracking (although it has not been fully developed in the new API by UPS yet)
- Cash on delivery

### CODE REVIEW

**Comment by manuelregidor** [comment]
@LoisRForgeFlow @rousseldenis @Cedric-Pigeon The previous version of the UPS connector, which used a different authentication method, had tests that set connection with UPS through credentials borrowed from an enterprise runbot. With oauth, that is not possible anymore, as there are not test credentials available for Odoo Enterprise. Should I use mock methods instead? Thank you.

**Comment by ValentinVinagre** [comment]
@pedrobaeza @etobella What do you think about this?

> @LoisRForgeFlow @rousseldenis @Cedric-Pigeon The previous version of the UPS connector, which used a different authentication method, had tests that set connection with UPS through credentials borrowed from an enterprise runbot. With oauth, that is not possible anymore, as there are not test credentials available for Odoo Enterprise. Should I use mock methods instead? Thank you.

**Comment by pedrobaeza** [comment]
OK, do the mocks.

**Comment by manuelregidor** [comment]
@LoisRForgeFlow @rousseldenis @Cedric-Pigeon Could you review, please? Thank you.

**Comment by rousseldenis** [comment]
> @LoisRForgeFlow @rousseldenis @Cedric-Pigeon Could you review, please? Thank you.

@manuelregidor Could you change PR's title ?

**Review by luis-ron — APPROVED** [review]
Functional review: LGTM 👍🏻

**Comment by luis-ron** [comment]
@ValentinVinagre @Tisho99 Could you review, please? Thank you.

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻 .
@JordiBForgeFlow @JordiMForgeFlow @LoisRForgeFlow 
you need finally this module? if yes, could do a review 😄

File: delivery_ups_oca/models/ups_request.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        self.shipper_number = self.carrier.ups_shipper_number
+        self.service_code = self.carrier.ups_service_code
+        self.file_format = self.carrier.ups_file_format
+        self.package_dimension_code = self.carrier.ups_package_dimension_code
+        self.package_weight_code = self.carrier.ups_package_weight_code
+        self.transaction_src = "Odoo (%s)" % self.carrier.name
+        self.client_id = self.carrier.ups_client_id
+        self.client_secret = self.carrier.ups_client_secret
+        self.token = self.carrier.ups_token
+        self.token_expipration_date = self.carrier.ups_token_expiration_date
```
Comment (JordiMForgeFlow) [inline delivery_ups_oca/models/ups_request.py]: ``
```suggestion
        self.token_expiration_date = self.carrier.ups_token_expiration_date
```

**Review by JordiMForgeFlow — COMMENTED** [review]
sorry for the nitpicking 🙏🏼, for the rest code review LGTM

File: delivery_ups_oca/models/ups_request.py (hunk 1 above)
Comment (manuelregidor) [reply delivery_ups_oca/models/ups_request.py]: @JordiMForgeFlow Done.

**Review by JordiMForgeFlow — APPROVED** [review]

**Comment by HaraldPanten** [comment]
@LoisRForgeFlow @JordiBForgeFlow @etobella  from the 1st of June, UPS will need this authentication. Could we move this forward?

**Pull request merged**